src/main/java/io/vertx/core/spi/json/JsonCodec.java

```diff
@@ -21,7 +21,15 @@
  */
 public interface JsonCodec {
 
-  JsonCodec INSTANCE = ServiceHelper.loadFactory(JsonCodec.class);
+  static JsonCodec loadCodec() {
+    JsonCodec instance = ServiceHelper.loadFactoryOrNull(JsonCodec.class);
```

Comment:
JsonCodec should have an `ordinal` method, and the returned instance should be the one with the highest ordinal. We may end up with multiple implementations on the classpath, and we need to have a deterministic way to retrieve always the same one.

Reply:
BTW, the same should be applied to the Vert.x 4 JsonFactory.

Reply:
One way would be to introduce a `Sorted` interface with a single `ordinal()` method, and in `io.vertx.core.ServiceHelper#loadFactories(java.lang.Class<T>, java.lang.ClassLoader)`, if the given interface implements `Sorted`, just sort the collections (reverse as we want highest ordinals first).

Reply:
we don't expect to have another implementation than the quarkus one, so I think it is not necessary for the moment.

Reply:
the point is that anyone would anyway use the same value to get higher priority at the end. I've seen similar things in the past and it never really worked as far as I could observe.

What we could do instead is something that applies all codecs and each codec will set itself as a codec unless there is already one codec here. The quarkus one would always set itself whatever the value is.

Reply:
Something like

```java
  default JsonCodec init(JsonCodec previous) {
    if (previous == null) {
      return this;
    } else {
      return previous;
    }
  }
```

that the load method will call. Then quarkus can have an implementation that always return this.